e2e_test/batch/functions/sqrt.slt.part

```diff
@@ -0,0 +1,14 @@
+query T
+SELECT abs(sqrt(1004.3) - 31.690692639953454) < 1e-12;
```

Comment:
Let's use an explicit cast to `double precision` or keep it quoted, so that it is testing 64-bit float rather than `numeric` (aka `decimal`).

FYI PostgreSQL's rules of typing:
```
select pg_typeof(sqrt('1004.3')), pg_typeof('1004.3')
union all
select pg_typeof(sqrt(1004.3)), pg_typeof(1004.3)
union all
select pg_typeof(sqrt(25)), pg_typeof(25)
;

    pg_typeof     | pg_typeof 
------------------+-----------
 double precision | unknown
 numeric          | numeric
 double precision | integer
(3 rows)
```

That is, there are 2 versions of `sqrt` function, one for `double precision` and one for `numeric`, as #8899 listed. In PostgreSQL, it prefers the `double precision` version unless the input is `numeric`, and `1004.3` without quotes is considered `numeric`.

Here we only have one version handling `double precision`, so it is chosen to handle `1004.3` as well. This test would be testing a different thing when we add the second version. And, we can now add `sqrt(numeric)` too if you'd like, given #9043 is merged.

Reply:
added support and test for `sqrt(numeric)`